elec/src/main-pcb.ato

```diff
@@ -0,0 +1,270 @@
+    second_en_button.output ~ gnd
+
+    # Add USB-C connector that can handle up to 5A at 20V (100W)
+    usb_c_connector = new USB_C_Connector_5A
+    usb_c_connector.usb ~ usb
+    usb_c_connector.usb_pd ~ usb_pd
+
+    # Add a TVS/ESD protection diode to protect from voltage spikes from the USB-C connector
+    esd_protection_diode = new SMBJ33CA
+    esd_protection_diode.power ~ usb_c_connector.power
+
+    # Add USB to serial via CP2102N
+    usb_to_serial = new CP2102N
+    usb_to_serial.uart.rx ~ uart.tx
+    usb_to_serial.uart.tx ~ uart.rx
+    usb_to_serial.usb ~ usb
+
+    # Add power mosfet circuit, so the CP2102N is turned on only if USB is connected
+    usbserial_power_mosfet_circuit = new MOSFETcircuitPowerOnOff_1x_1A
+    usbserial_power_mosfet_circuit.power_in ~ power_3_3v
+    usbserial_power_mosfet_circuit.power_out ~ usb_to_serial.power
+    usbserial_power_mosfet_circuit.control ~ usb_c_connector.power.vcc
+
+    # Microphone
+    microphone = new MSM261D4030H1CPM
+    microphone.i2s ~ i2s
+    microphone.power.gnd ~ gnd
+
+    # Add a slider switch to turn on/off the microphone
+    microphone_power_slider_switch = new SK12D07VG4
+    microphone_power_slider_switch.p2 ~ microphone.power.vcc
+    microphone_power_slider_switch.p3 ~ power_3_3v.vcc
+    microphone_power_slider_switch.gnd ~ gnd
+
+    # Add a slider to switch between GlowLight & GlowTower mode (which sets number of leds, wifi name, etc.)
+    lampmode_slider_switch = new SK12D07VG4
+    lampmode_slider_switch.p1 ~ gnd
+    lampmode_slider_switch.p2 ~ esp32.ic.IO35
+    lampmode_slider_switch.p3 ~ power_3_3v.vcc
+    lampmode_slider_switch.gnd ~ gnd
+
+    # Add a pullup resistor to the slider switch, so it's not floating
+    lampmode_slider_pullup_resistor = new Resistor
+    lampmode_slider_pullup_resistor.resistance = 100kohm +/- 1%
+    lampmode_slider_pullup_resistor.footprint = "R0402"
+    lampmode_slider_pullup_resistor.mpn = "C25741"
+    lampmode_slider_pullup_resistor.p1 ~ esp32.ic.IO35
+    lampmode_slider_pullup_resistor.p2 ~ power_3_3v.vcc
+
+    # Pin headers, 2.54mm pitch
+    pin_header_male_led_strip_1 = new PinHeader_2_54mm_3pins
+    pin_header_male_led_strip_1.p3 ~ gnd
+
+    pin_header_male_led_strip_2 = new PinHeader_2_54mm_3pins
+    pin_header_male_led_strip_2.p3 ~ gnd
+
+    pin_header_male_led_strip_3 = new PinHeader_2_54mm_3pins
+    pin_header_male_led_strip_3.p3 ~ gnd
+
+    pin_header_male_led_strip_4 = new PinHeader_2_54mm_3pins
+    pin_header_male_led_strip_4.p3 ~ gnd
+
+    # add the pin headers for the action button PCB that also includes an LED ring (as a status indicator for the user)
+    pin_header_male_action_button = new PinHeader_2_54mm_2pins
+    pin_header_male_action_button.p1 ~ gnd
+    pin_header_male_action_button.p2 ~ esp32.ic.IO0
+
+    pin_header_male_led_ring = new PinHeader_2_54mm_2pins
+
+    # LED levelshifters
+    levelshifter_1 = new SN74LVC2T45DCUR                                # for LED strip 1 + 2
```

Comment:
I am not taking any chances with the levelshifter, since it needs to be fast enough for WS2812B LEDs, if I recall the WLED documentation correctly. So I only want to use levelshifters here which are confirmed by others to work with WS2812B LEDs. And since I tested that component in the past already, it's probably better to play it safe here I think.

Reply:
Sure no worries, I have used it on a WS2812 here https://gitlab.atopile.io/community-projects/blinky with success. Transistors are pretty fast ;)

No need to change, just a thought.

Reply:
Oh interesting. But in the atopile file it just says import level shifter - but which one do you use?